ui/app/routes/access-control.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Route from '@ember/routing/route';
+import withForbiddenState from 'nomad-ui/mixins/with-forbidden-state';
+import WithModelErrorHandling from 'nomad-ui/mixins/with-model-error-handling';
+import { inject as service } from '@ember/service';
+import RSVP from 'rsvp';
+
+export default class AccessControlRoute extends Route.extend(
+  withForbiddenState,
+  WithModelErrorHandling
+) {
+  @service can;
+  @service store;
+  @service router;
+
+  beforeModel() {
+    // TODO: policy listing is currently selfTokenIsManagement dependent.
+    // If this should ever change and become its own ACL permission, we should
+    // expand this to include roles and tokens checks as well. Otherwise, they're
+    // all the same permission.
+    if (this.can.cannot('list policies')) {
+      this.router.transitionTo('/jobs');
+    }
+  }
+
+  // Load our tokens, roles, and policies
+  model() {
+    return RSVP.hash({
+      tokens: this.store.findAll('token'),
+      roles: this.store.findAll('role'),
+      policies: this.store.findAll('policy'),
+    });
+  }
```

Comment:
This is the main entry-point for all the new sub-pages. As such, we pre-load all the relevant data up front.

There's a world where we may have to paginate this in the future, but for now this approach is sensible.

Reply:
Won't this call end up being very slow?

Reply:
Not too badly, in testing. But even if it did: this is an administrative page with deliberate inter-twining of the entities we load. I wouldn't want to emphasize performance at the cost of full control here, like I might want to do in a more general-use / read-only page.